### PR TITLE
[Enhancement] Support infer struct type from parquet in files()

### PR DESCRIPTION
**Pull request by wyb**
## Why I'm doing:
```
mysql> select * from  files("path" = "hive_struct_timestamp0.parquet", "format" = "parquet";
ERROR 1064 (HY000): arrow type 18 is not supported
```

`struct<a datetime>` is not supported

struct type is infered to varchar or json type.

## What I'm doing:
support struct type

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/parquet_schema_builder.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@ static Status get_parquet_type_from_group(const ::parquet::schema::NodePtr& node
 static Status get_parquet_type_from_primitive(const ::parquet::schema::NodePtr& node, TypeDescriptor* type_desc);
 static Status get_parquet_type_from_list(const ::parquet::schema::NodePtr& node, TypeDescriptor* type_desc);
 static Status get_parquet_type_from_map(const ::parquet::schema::NodePtr& node, TypeDescriptor* type_desc);
+static Status try_to_infer_struct_type(const ::parquet::schema::NodePtr& node, TypeDescriptor* type_desc);
```
Comment (jaogoy) [inline be/src/exec/parquet_schema_builder.cpp:26]: Why not to keep the code style to use `get_parquet_type_from_struct`?

File: be/src/exec/parquet_schema_builder.cpp [hunk 2]
```diff
@@ -217,4 +223,42 @@ static Status get_parquet_type_from_map(const ::parquet::schema::NodePtr& node,
     return Status::OK();
 }
 
+/*
+try to infer struct type from group node.
+
+group <name> {
+    type field0;
```
Comment (jaogoy) [inline be/src/exec/parquet_schema_builder.cpp:230]: Where is the definition from?

Comment (wyb) [reply be/src/exec/parquet_schema_builder.cpp:230]: parquet does not have struct type, there is no struct definition in parquet, so i use `infer struct type`.

Comment (wyb) [reply be/src/exec/parquet_schema_builder.cpp:232]: add comment

**Review by jaogoy — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Review by Smith-Cruise — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-3.2

**Pull request merged**